### PR TITLE
refactor config values management

### PR DESCRIPTION
**Pull request by matheusalcantarazup**
**- What I did**
This commit refactor the config values management to be more clear when
values are filled via command line flags, configuration file or using
environment variables and what are the default values.
The `config.IConfig` interface expose getters and setter to read and set
values on config instance.  These getters are confuse, because some of
them return the value informed by user or the default value and others
just return the value informed by user, which make difficult to inspect
all default values from config. These setters are used to override the
config values from configuration file and environment variables, but they
are confuse because in multiples places we are changing the configuration
that can leave in a inconsistent state. For example, the start and
generate command, call functions to override the config values to the
values from configuration file and environment variables.
So, this commit improve the config management to load the default
values, read the user input from command lines, configuration file and
environment variable in on single place, and change the commands to
receive a "ready to use" config. This commit also remove the
`config.IConfig` interface to make more simple.
NOTE: We still have getters and setters on config struct because we call
theses functions in several places, so we need to remove them and change
the references to use the public fields from config.

Co-authored-by: Matheus Alcantara <matheus.alcantara@zup.com.br>

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: cmd/app/start/start.go [hunk 1]
```diff
@@ -92,76 +86,223 @@ func (s *Start) CreateStartCommand() *cobra.Command {
...
-		StringP("custom-rules-path", "c", s.configs.GetContainerBindProjectPath(), "Used to pass the path to the horusec custom rules file. Example: -c=\"./horusec/horusec-custom-rules.json\".")
-	_ = startCmd.PersistentFlags().
-		BoolP("information-severity", "I", s.configs.GetEnableInformationSeverity(), "Used to enable or disable information severity vulnerabilities, information vulnerabilities can contain a lot of false positives. Example: -I=\"true\"")
-	_ = startCmd.PersistentFlags().
-		StringSliceP("show-vulnerabilities-types", "", s.configs.GetShowVulnerabilitiesTypes(), "Used to show in the output vulnerabilities of types: Vulnerability, Risk Accepted, False Positive, Corrected. Example --show-vulnerabilities-types=\"Vulnerability, Risk Accepted\"")
-	_ = startCmd.PersistentFlags().
-		BoolP("enable-owasp-dependency-check", "w", s.configs.GetEnableOwaspDependencyCheck(), "Enable owasp dependency check. Example -w=\"true\". Default: false")
-	_ = startCmd.PersistentFlags().
-		BoolP("enable-shellcheck", "j", s.configs.GetEnableShellCheck(), "Enable shellcheck. Example -h=\"true\". Default: false")
+
```
Comment (lucasbrunozup) [inline cmd/app/start/start.go:89]: Much better!

File: config/config.go [hunk 2]
```diff
@@ -16,686 +16,634 @@ package config
...
+	}
 
-	return c
+	return &Config{
+		Version: version,
+		GlobalOptions: GlobalOptions{
+			ConfigFilePath: filepath.Join(wd, "horusec-config.json"),
+			LogLevel:       logrus.InfoLevel.String(),
+			LogFilePath: filepath.Join(
+				os.TempDir(), fmt.Sprintf("horusec-log-%s.log", time.Now().Format("2006-01-02 15:04:05")),
```
Comment (matheusalcantarazup) [inline config/config.go:144]: NOTE: I changed the log path to be created on temp directory instead current working directory. I think that is better create log files on temp directory instead messy the directory of use with a lot of log files.

**Comment by lucasbrunozup** [comment]
Excellent changes will be of great importance. 

Could you rename the branch name to our pattern, please?
feature/**  or  bugfix/** or chore/**

**Comment by matheusalcantarazup** [comment]
Yes, I can rename the branch name, but whats the problem with current name? This name will not be added on commit message, and the branch will be deleted.

**Comment by lucasbrunozup** [comment]
It's the convention we currently have for naming our project's branches.

**Review by nathanmartinszup — APPROVED** [review]

**Pull request merged**